crates/testlang/outputs/cargo/tests/src/versions/mod.rs

```diff
@@ -1,11 +1,11 @@
 use semver::Version;
-use slang_solidity::language::Language;
+use slang_testlang::language::Language;
 
 #[test]
```

Comment:
I wonder why these tests were moved out of `Solidity`?
For example, this one in perticular asserts the minimum solidity supported version, which is a user-facing piece of information.

Reply:
If the intent is for the tests in `testlang` to expand to test other Slang-properties (not specific to Solidity), then I suggest (at least for now) to just copy the tests instead of moving them, and they can evolve separately from there.

Reply:
Because IMO they add no real value that is solidity specific. They have no real coverage, and if they did they would merely replicate the data. They merely test the versions mechanism, which is not solidity specific.

To test the minimum version you would need to test every possible version before 0.4.11 to verify that it is the minimum. Otherwise it is only testing that 0.4.11 is included, which is in the data.

Reply:
I agree in principle. The tests make sure we do generate a valid sorted `SUPPORTED_VERSIONS`, which can be tested in `testlang`. But practically speaking, given that (excluding CST snapshots), we have less than 15 unit tests covering the entire public API, which barely qualifies as a "sanity check". Given how little effort it takes to maintain them (so far), I think it is worth it to keep them.

While we have some degree of confidence in our build/codegen pipeline that it will maintain both languages in check, accidental divergence and bugs can and will happen (I just discovered a tiny one while reviewing this PR: #812), and the tests will prevent "cross contamination".

Reply:
Duplicated tests are the same as one test. So keeping the tests in the solidity crates/package adds nothing to test coverage or test value. IMO we should eliminate the reduncdancy that adds no actual value. To think that it adds to our test coverage is wrong, and serves merely to mislead us.

And I strongly feel that testing in the solidity crate should be about the grammar, and other things that are entirely specific to the language. Everything else about slang should be tested in the non-solidity crates/packages.

Relying on some random test duplication to detect possible divergence in the tiny part of overlap is also misleading security.

Lets F2F this.